droid-results/src/main/java/uk/gov/nationalarchives/droid/profile/CsvItemWriter.java

```diff
@@ -220,7 +256,43 @@ public void open(final Writer writer) {
         if (headers == null) {
             headers = Arrays.copyOf(HEADERS, HEADERS.length) ;
         }
-        String[] headersToWrite = getHeadersToWrite(headers);
+    }
+
+    private void writeHeadersForOneLinePerFormatExport(List<? extends ProfileResourceNode> nodes) {
+
+        if (options == ExportOptions.ONE_ROW_PER_FORMAT) {
+            throw new RuntimeException("Unexpectedly called per file header creation. Unable to proceed");
+        }
+
+        List<String> headersToWrite = Arrays.stream(getHeadersToWrite(headers)).collect(Collectors.toList()) ;
+
+        //if we are writing one row per file, then we tag the "per format" fields as additional columns,
+        //if such columns need to be added, we create appropriate headers with a running suffix and write
+        //them to the file
+        if (!Collections.disjoint(headersToWrite, PER_FORMAT_HEADERS)) {
+            Optional<Integer> maxIdentificationsOption = nodes.stream().map(ProfileResourceNode::getIdentificationCount).collect(Collectors.toList()).stream().filter(Objects::nonNull).max(Integer::compare);
+            int maxIdentifications = 0;
+            if (maxIdentificationsOption.isPresent()) {
+                maxIdentifications = maxIdentificationsOption.get();
+            }
+            if (maxIdentifications > 1) { //add headers
+                for (int newColumnSuffix = 2; newColumnSuffix <= maxIdentifications; newColumnSuffix++) {
+                    //"PUID","MIME_TYPE","FORMAT_NAME","FORMAT_VERSION"
+                    if (headersToWrite.contains(HEADER_NAME_PUID)) {
+                        headersToWrite.add(HEADER_NAME_PUID + newColumnSuffix);
+                    }
+                    if (headersToWrite.contains(HEADER_NAME_MIME_TYPE)) {
+                        headersToWrite.add(HEADER_NAME_MIME_TYPE + newColumnSuffix);
+                    }
+                    if (headersToWrite.contains(HEADER_NAME_FORMAT_NAME)) {
+                        headersToWrite.add(HEADER_NAME_FORMAT_NAME + newColumnSuffix);
+                    }
+                    if (headersToWrite.contains(HEADER_NAME_FORMAT_VERSION)) {
+                        headersToWrite.add(HEADER_NAME_FORMAT_VERSION + newColumnSuffix);
+                    }
```

Comment:
Is it worth putting another loop here to iterate over `PER_FORMAT_HEADERS` to reduce the duplication? Readability would take a hit.